test/gtest/uct/test_peer_failure.cc

```diff
@@ -538,3 +540,68 @@ UCS_TEST_SKIP_COND_P(test_uct_peer_failure_multiple, test,
 }
 
 UCT_INSTANTIATE_TEST_CASE(test_uct_peer_failure_multiple)
+
+class test_uct_peer_failure_keepalive : public test_uct_peer_failure
+{
+protected:
+    void is_skipped()
+    {
+        if (!has_transport("rc_mlx5")) {
+            UCS_TEST_SKIP_R("Unsupported");
+        }
+    }
+};
+
+UCS_TEST_P(test_uct_peer_failure_keepalive, killed)
+{
+    ucs_status_t status;
+
+    is_skipped();
+
+    scoped_log_handler slh(wrap_errors_logger);
+    flush();
+    EXPECT_EQ(0, m_err_count); /* it works !!! */
+    flush();
+
+    status = uct_ep_flush(ep0(), UCT_FLUSH_FLAG_CHECK, NULL);
+    ASSERT_UCS_OK(status);
+    flush();
+
+    kill_receiver();
+
+    status = uct_ep_flush(ep0(), UCT_FLUSH_FLAG_CHECK, NULL);
+    ASSERT_UCS_OK(status);
+    flush();
+
+    EXPECT_EQ(1, m_err_count);
+}
+
+UCS_TEST_P(test_uct_peer_failure_keepalive, resources)
+{
+    /* test for flush(CHECK) does nothing if there are TX operations
+     * scheduled */
+    ucs_status_t status;
+
+    is_skipped();
+
+    scoped_log_handler slh(wrap_errors_logger);
+
+    /* hack: send dummy data, but don't progress iface/ep. In this case UCT
+     * will not process completions and 'send' operation will be in state
+     * 'scheduled' even in case if it is completed on HW. In this state
+     * flush(CHECK) should not make any actions and just returns OK status.
+     */
+    send_am(0);
+    kill_receiver();
+    /* after kill_receiver all send/recv operations will fail. flush(CHECK)
+     * should not make anything and no errors should be registered */
+
+    status = uct_ep_flush(ep0(), UCT_FLUSH_FLAG_CHECK, NULL);
+    ASSERT_UCS_OK(status);
+    flush();
+
+    EXPECT_EQ(0, m_err_count);
+}
+
+/* TODO: add new transports when implemented */
+UCT_INSTANTIATE_IB_TEST_CASE(test_uct_peer_failure_keepalive)
```

Comment:
can instantiate for rc_mlx5 only for now

Reply:
done